CMakeLists.txt

```diff
@@ -29,114 +33,21 @@ if (BUILD_TESTING)
-    src/modules/location/Location.cpp
-    src/modules/lorem/Lorem.cpp
-    src/modules/medicine/Medicine.cpp
-    src/modules/movie/Movie.cpp
-    src/modules/music/Music.cpp
-    src/modules/person/Person.cpp
-    src/modules/phone/Phone.cpp
-    src/modules/plant/Plant.cpp
-    src/modules/science/Science.cpp
-    src/modules/sport/Sport.cpp
-    src/modules/string/String.cpp
-    src/modules/system/System.cpp
-    src/modules/vehicle/Vehicle.cpp
-    src/modules/videoGame/VideoGame.cpp
-    src/modules/weather/Weather.cpp
-    src/modules/word/Word.cpp
-    src/common/FormatHelper.cpp
-    src/common/LuhnCheck.cpp
-    src/common/StringHelper.cpp
-)
+add_library(${CMAKE_PROJECT_NAME})
 
-add_library(${CMAKE_PROJECT_NAME} ${FAKER_SOURCES})
-
-target_include_directories(
-    ${CMAKE_PROJECT_NAME} PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    $<INSTALL_INTERFACE:include>
-)
-
-target_include_directories(
-    ${CMAKE_PROJECT_NAME} PRIVATE
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
-)
-
-target_compile_features(${CMAKE_PROJECT_NAME} PUBLIC cxx_std_20)
-configure_compiler_warnings(${CMAKE_PROJECT_NAME}
-    "${WARNINGS_AS_ERRORS}"
-    "${MSVC_WARNINGS}"
-    "${CLANG_WARNINGS}"
-    "${GCC_WARNINGS}")
-
-include(GenerateExportHeader)
-generate_export_header(${CMAKE_PROJECT_NAME} BASE_NAME FAKER_CXX EXPORT_FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/include/faker-cxx/Export.h)
-set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES
-    CXX_VISIBILITY_PRESET hidden
-    CMAKE_VISIBILITY_INLINES_HIDDEN 1
-)
-
-target_include_directories(
-    ${CMAKE_PROJECT_NAME}
-    PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
-    PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/include
-)
-
-include(GNUInstallDirs)
-install(TARGETS ${CMAKE_PROJECT_NAME}
-    EXPORT ${CMAKE_PROJECT_NAME}-targets
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
-
-install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/faker-cxx
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-    FILES_MATCHING
-    PATTERN "*.h"
-)
-
-install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/include/faker-cxx
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-    FILES_MATCHING
-    PATTERN "*.h"
-)
-
-install(EXPORT ${CMAKE_PROJECT_NAME}-targets
-    NAMESPACE ${CMAKE_PROJECT_NAME}::
-    FILE ${CMAKE_PROJECT_NAME}-config.cmake
-    DESTINATION lib/cmake/${CMAKE_PROJECT_NAME})
-
-if (USE_STD_FORMAT)
-    target_compile_definitions(${CMAKE_PROJECT_NAME} PRIVATE USE_STD_FORMAT)
-elseif (USE_SYSTEM_DEPENDENCIES)
+if (USE_SYSTEM_DEPENDENCIES)
     find_package(fmt CONFIG REQUIRED)
-    target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE fmt::fmt)
-else ()
-    add_subdirectory(externals/fmt)
+elseif (NOT USE_STD_FORMAT)
+    find_library(FMT_LIBRARY fmt)
+    if (NOT FMT_LIBRARY)
+        message(FATAL_ERROR "Could not find fmt library. Please, read the contribution guide.")
```

Comment:
Added fatal error message in case not finding fmt library and is not possible to use std::format. Hope users can be guided by contributor guide.